File: tests/testdata/codegen/deferred-outputs-pp/nodejs/deferred-outputs.ts
import * as codeinfra from "@codeinfra/codeinfra";
import { First } from "./first";
import { Second } from "./second";

const [secondPasswordLength, resolveSecondPasswordLength] = codeinfra.deferredOutput<number>();
const first = new First("first", {passwordLength: secondPasswordLength});
const second = new Second("second", {petName: first.petName});
resolveSecondPasswordLength(second.passwordLength);
const [loopingOverMany, resolveLoopingOverMany] = codeinfra.deferredOutput<Array<number>>();
const another = new First("another", {passwordLength: loopingOverMany.apply(loopingOverMany => loopingOverMany.length)});
const many: Second[] = [];
for (const range = {value: 0}; range.value < 10; range.value++) {
    many.push(new Second(`many-${range.value}`, {petName: another.petName}));
}
resolveLoopingOverMany(codeinfra.output(many.map((v, k) => [k, v]).map(([_, v]) => (v.passwordLength))));

File: tests/testdata/codegen/deferred-outputs-pp/dotnet/deferred-outputs.cs
using System.Collections.Generic;
using System.Linq;
using Codeinfra;

return await Deployment.RunAsync(() => 
{
    var secondPasswordLength = new Codeinfra.DeferredOutput<int>();
    var first = new Components.First("first", new()
    {
        PasswordLength = secondPasswordLength.Output,
    });

    var second = new Components.Second("second", new()
    {
        PetName = first.PetName,
    });

    secondPasswordLength.Resolve(second.PasswordLength);
    var loopingOverMany = new Codeinfra.DeferredOutput<List<int>>();
    var another = new Components.First("another", new()
    {
        PasswordLength = loopingOverMany.Output.Apply(loopingOverMany => loopingOverMany.Length),
    });

    var many = new List<Components.Second>();
    for (var rangeIndex = 0; rangeIndex < 10; rangeIndex++)
    {
        var range = new { Value = rangeIndex };
        many.Add(new Components.Second($"many-{range.Value}", new()
        {
            PetName = another.PetName,
        }));
    }
    loopingOverMany.Resolve(Output.Create(many.Select((value, i) => new { Key = i.ToString(), Value = pair.Value }).Select(v => 
    {
        return v.PasswordLength;
    }).ToList()));
});

